activerecord/lib/active_record/connection_adapters/mysql2/database_statements.rb

```diff
@@ -84,6 +72,10 @@ def perform_query(raw_connection, sql, binds, type_casted_binds, prepare:, notif
             verified!
             handle_warnings(sql)
             result
+          ensure
+            if batch && !multi_statements_enabled? && active?
```

Comment:
I think I would prefer that this used a local, set in the `if` above, rather than independently working out whether it thinks we would have `set_server_option`

Reply:
You mean flip a flag in:

```ruby
            if batch && !multi_statements_enabled?
              raw_connection.set_server_option(::Mysql2::Client::OPTION_MULTI_STATEMENTS_ON)
            end
```
?

Reply:
Yeah, just to avoid duplicating the [multi-factor] conditional

Reply:
Makes sense.